protocol/x/sending/app_test.go

```diff
@@ -773,3 +778,203 @@ func getSubaccountAssetQuantums(
 	}
 	return big.NewInt(0) // by default, subaccount has 0 of this `asset`.
 }
+
+func TestWithdrawalGating_ChainOutage(t *testing.T) {
+	tests := map[string]struct {
+		// State.
+		subaccount satypes.Subaccount
+
+		// Parameters.
+		secondsBetweenBlocks uint32
+
+		// Configuration.
+		isWithdrawal bool
+
+		// Expectations.
+		expectedWithdrawalsGated bool
+	}{
+		`5 minutes passes between blocks and withdrawals are gated after the chain restarts`: {
+			subaccount: constants.Dave_Num1_10_000USD,
+
+			secondsBetweenBlocks: 60 * 5,
```

Comment:
Both of these durations (5m, 30m) are set in the [default genesis](https://github.com/dydxprotocol/v4-chain/blob/0c4e1d9771d388fd547e8f1fb2c5a1b41b98d79b/protocol/x/blocktime/types/genesis.go#L8-L12).